File: DayBagBuilder/Models/BagItem.cs
using System;
using System.Collections.Generic;

namespace DayBagBuilder.Models
{
    public partial class BagItem
    {
        public int Id { get; set; }
        public string? ItemName { get; set; }
        public bool? ForSunny { get; set; }
        public bool? ForHot { get; set; }
        public bool? ForCold { get; set; }
        public bool? ForRain { get; set; }
        public bool? ForSnow { get; set; }
        public int? Itemweight { get; set; }
    }
}

File: DayBagBuilder/ClientApp/src/app/BagItem.ts
export class BagItem{
    id: number;
    itemName: string;
    forSunny: boolean;
    forHot: boolean;
    forCold: boolean;
    forRain: boolean;
    forSnow: boolean;
    itemweight: number;

    constructor(id: number, itemName: string, forSunny: boolean, forHot: boolean, forCold: boolean, forRain: boolean, forSnow: boolean, itemweight: number){
        this.id = id;
        this.itemName = itemName;
        this.forSunny = forSunny;
        this.forHot = forHot;
        this.forCold = forCold;
        this.forRain = forRain;
        this.forSnow = forSnow;
        this.itemweight = itemweight;
    }
}